float32 can't compare to math.MaxFloat32

The following program `sample.go` triggers an error:

```go
package main

import "math"
    
func main() {
    var f float32
    if f < math.MaxFloat32 {
        println("ok")
    }
}
```

Expected result:
```console
$ go run ./sample.go
ok
```

Got:
```console
$ yaegi ./sample.go
./sample.go:7:5: mismatched types .float32 and .float64
```

--- a/_test/math1.go
+++ b/_test/math1.go
@@ -1,0 +1,13 @@
+package main
+
+import "math"
+
+func main() {
+	var f float32
+	if f < math.MaxFloat32 {
+		println("ok")
+	}
+}
+
+// Output:
+// ok

--- a/cmd/goexports/goexports.go
+++ b/cmd/goexports/goexports.go
@@ -254,7 +254,14 @@ func genContent(dest, pkgName, license string) ([]byte, error) {
 
 // fixConst checks untyped constant value, converting it if necessary to avoid overflow
 func fixConst(name string, val constant.Value) string {
-	if val.Kind() == constant.Int {
+	switch val.Kind() {
+	case constant.Float:
+		str := val.ExactString()
+		if _, err := strconv.ParseFloat(str, 32); err == nil {
+			return "float32(" + name + ")"
+		}
+		return name
+	case constant.Int:
 		str := val.ExactString()
 		i, err := strconv.ParseInt(str, 0, 64)
 		if err == nil {
@@ -271,8 +278,10 @@ func fixConst(name string, val constant.Value) string {
 		if err == nil {
 			return "uint64(" + name + ")"
 		}
+		return name
+	default:
+		return name
 	}
-	return name
 }
 
 // genLicense generates the correct LICENSE header text from the provided

--- a/stdlib/go1_12_math.go
+++ b/stdlib/go1_12_math.go
@@ -61,7 +61,7 @@ func init() {
 		"Log2E":                  reflect.ValueOf(math.Log2E),
 		"Logb":                   reflect.ValueOf(math.Logb),
 		"Max":                    reflect.ValueOf(math.Max),
-		"MaxFloat32":             reflect.ValueOf(math.MaxFloat32),
+		"MaxFloat32":             reflect.ValueOf(float32(math.MaxFloat32)),
 		"MaxFloat64":             reflect.ValueOf(math.MaxFloat64),
 		"MaxInt16":               reflect.ValueOf(math.MaxInt16),
 		"MaxInt32":               reflect.ValueOf(math.MaxInt32),

--- a/stdlib/go1_13_math.go
+++ b/stdlib/go1_13_math.go
@@ -61,7 +61,7 @@ func init() {
 		"Log2E":                  reflect.ValueOf(math.Log2E),
 		"Logb":                   reflect.ValueOf(math.Logb),
 		"Max":                    reflect.ValueOf(math.Max),
-		"MaxFloat32":             reflect.ValueOf(math.MaxFloat32),
+		"MaxFloat32":             reflect.ValueOf(float32(math.MaxFloat32)),
 		"MaxFloat64":             reflect.ValueOf(math.MaxFloat64),
 		"MaxInt16":               reflect.ValueOf(math.MaxInt16),
 		"MaxInt32":               reflect.ValueOf(math.MaxInt32),

